src/Microsoft.Health.Fhir.Azure/IntegrationDataStore/AzureBlobSourceStream.cs

```diff
@@ -172,7 +172,7 @@ private async Task RefreshBlobClientAsync()
 
         private async Task<Stream> DownloadDataFunc(long offset, long length)
         {
-            var stream = new RecyclableMemoryStream(_recyclableMemoryStreamManager);
+            var stream = new RecyclableMemoryStream(_recyclableMemoryStreamManager, tag: nameof(AzureBlobSourceStream));
```

Comment:
Wouldn't you want to use a "using" statement here as well? No guarantees when garbage collection would get to it.

Reply:
Nevermind, I see it gets returned.